src/rules/require-source-directory.ts

```diff
@@ -0,0 +1,24 @@
+import { buildRule } from './helpers';
+import { RuleName } from './types';
+
+export default buildRule({
+  name: RuleName.RequireSourceDirectory,
+  description: 'Does the `src/` directory exist?',
+  dependencies: [],
+  execute: async ({ project, pass, fail }) => {
+    const entryPath = 'src';
+
+    const stats = await project.fs.getEntryStats(entryPath);
+
+    if (stats?.isDirectory()) {
+      return pass();
+    }
+
+    const message =
+      stats && !stats.isDirectory()
+        ? `\`${entryPath}/\` is not a directory when it should be.`
+        : `\`${entryPath}/\` does not exist in this project.`;
+    const failures = [{ message }];
+    return fail(failures);
```

Comment:
Nit: We could eliminate the extra check by restructing it like this:

```suggestion
    if (!stats) {
      return fail([{ message: `\`${entryPath}/\` does not exist in this project.` }]);
    } else if (!stats.isDirectory()) {
      return fail([{ message: `\`${entryPath}/\` is not a directory when it should be.` }]);
    }

    return pass();
```

Or maybe this if we want to avoid repeating the `fail` call:

```suggestion
    const failures = [];
    if (!stats) {
      failures.push({ message: `\`${entryPath}/\` does not exist in this project.` });
    } else if (!stats.isDirectory()) {
      failures.push({ message: `\`${entryPath}/\` is not a directory when it should be.` }});
    }
    
    if (failures.length > 0) {
      return fail(failures);
    }
    return pass();
```